chapters/object_types/samplers.txt

```diff
@@ -142,18 +142,24 @@ in column-major memory layout and are applied to column vectors
 
 Extension: `KHR_SAMPLER_PRIMITIVE`
 
-The primitive sampler samples an `Array1D` at unnormalized integer
-coordinates based on the `primitiveId` surface attribute, shifted by
-non-negative `offset`.
+The primitive sampler samples an `Array1D` at integer
+coordinates based on the `primitiveId` surface attribute.
+The `offset` parameter value is added to `primitiveID` before sampling.
 
 .Parameters accepted by all `primitive` samplers.
 [cols="<,<,<2",options="header,unbreakable"]
 |===================================================================================================
-| Name  | Type      | Description
-| array | `ARRAY1D` | backing array of the sampler
+| Name  | Type      | Default  | Description
+| array | `ARRAY1D` of `FLOAT32`/`FLOAT32_VEC2`/`FLOAT32_VEC3`/`FLOAT32_VEC4` |  | backing array of the sampler
```

Comment:
I remember we wanted to go for `ARRAY1D` of *Color*, didn't we? Because one common usecase is to use the sampler for per-primitive colors.

Reply:
Updated in below commit